modules/command-engine/core/src/main/scala/surge/exceptions/KafkaPublishTimeoutException.scala

```diff
@@ -3,4 +3,4 @@
 package surge.exceptions
 
 case class KafkaPublishTimeoutException(aggregateId: String, underlyingException: Throwable)
-    extends RuntimeException(s"Aggregate $aggregateId timed out trying to publish to Kafka", underlyingException)
+    extends RuntimeException(s"Aggregate $aggregateId timed out trying to publish to Kafka")
```

Comment:
During the testing, I've found that nested exceptions can't be serialized/deserialized. That's why I've removed it.

Reply:
Is that just for our exceptions? Or any exceptions (i.e. if the domain code throws a nested exception)? It seems unfortunate we'd lose all that additional context since the actual root cause is usually that `underlyingException`

Reply:
I did some testing. And found that the issue was the **underlyingException** field in KafkaPublishTimeoutException. Renamed it to the **cause**.
Now I think it should be good.